client/state/sites/domains/types.ts

```diff
@@ -10,7 +12,7 @@ export interface SiteDomain {
 	expired?: boolean;
 	expiry?: string | null;
 	expirySoon?: boolean;
-	googleAppsSubscription?: { status?: string };
+	googleAppsSubscription?: GoogleEmailSubscription | null;
 	titanMailSubscription?: { status?: string; appsUrl?: string };
```

Comment:
It may also be worth updating `titanMailSubscription` to be `TitanEmailSubscription | null`, but that's not blocking.